client/src/components/Meetup/Meetup.js

```diff
@@ -1,8 +1,689 @@
+    color: "grey",
+  },
+  tagList: {
+    padding: 0,
+    listStyle: "none",
+  },
+  tag: {
+    display: "inline",
+    margin: theme.spacing(0.25),
+  },
+  description: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+  attendeeList: {
+    marginTop: theme.spacing(2),
+    padding: 0,
+    listStyle: "none",
+  },
+  attendee: {
+    marginBottom: theme.spacing(1),
+  },
+  organizer: {
+    marginTop: theme.spacing(2),
+    width: 500,
+  },
+  organizerAvatar: {
+    width: theme.spacing(8),
+    height: theme.spacing(8),
+  },
+  organizerBio: {
+    overflow: "hidden",
+    textOverflow: "ellipsis",
+    display: "-webkit-box",
+    "-webkit-line-clamp": 3,
+    "-webkit-box-orient": "vertical",
+  },
+  avatar: {
+    width: theme.spacing(8),
+    height: theme.spacing(8),
+  },
+  profileName: {
+    textDecoration: "none",
+    color: theme.palette.common.black,
+  },
+  spinner: {
+    marginTop: theme.spacing(1),
+    marginBottom: theme.spacing(5),
+  },
+  error: {
+    color: theme.palette.error.main,
+  },
+  ownerInterests: {
+    display: "inline-block",
+    paddingLeft: 0,
+    paddingRight: theme.spacing(1),
+    listStyle: "none",
+  },
+  noInterests: {
+    padding: 0,
+    color: theme.palette.warning.main,
+  },
+}));
 function Meetup({ id }) {
-  return <>Information about meetup with id {id}</>;
+  const classes = useStyles();
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [isUpdating, setIsUpdating] = useState(false);
+  const [user, setUser] = useState(null);
+  const [showLogin, setShowLogin] = useState(false);
+  const [fetchMeetupError, setFetchMeetupError] = useState(null);
+  const [userMeetupStatus, setUserMeetupStatus] = useState("");
+  const [eventDetails, setEventDetails] = useState(null);
+  const [attendees, setAttendees] = useState([]);
+  const [pendingAttendees, setPendingAttendees] = useState([]);
+
+  const locale = "en-US";
+  const meetupTimeOptions = {
+    hour: "numeric",
+    minute: "numeric",
+    day: "numeric",
+    month: "long",
+    year: "numeric",
+    timeZoneName: "short",
+  };
+  const userDateOptions = {
+    day: "numeric",
+    month: "long",
+    year: "numeric",
+  };
+
+  async function setAttendeeLists(attendeeList, pendingAttendeesList) {
+    setAttendees(await fetcher.getDataForUsers(attendeeList));
+    setPendingAttendees(await fetcher.getDataForUsers(pendingAttendeesList));
+  }
+
+  const setUserStatus = (userData, meetupData) => {
+    if (userData && meetupData.owner === userData.id) {
+      setUserMeetupStatus("owner");
+    } else if (
+      meetupData.pendingAttendees &&
+      meetupData.pendingAttendees.includes(userData.id)
+    ) {
+      setUserMeetupStatus("pending");
+    } else if (
+      meetupData.attendees &&
+      meetupData.attendees.includes(userData.id)
+    ) {
+      setUserMeetupStatus("attending");
+    } else if (meetupData.rejected) {
+      setUserMeetupStatus("rejected");
```

Comment:
Can we make these statuses into constants and import them from the constants file?